TRDataControl/Transport/TRDataImporter.cs

```diff
@@ -118,6 +118,17 @@ private void CleanRemovalList()
                 cleanedEntities.Add(type);
             }
         }
+
+        for (int i = cleanedEntities.Count - 1; i >= 0; i--)
```

Comment:
This seems safer and easier to understand than the reverse iteration through a list that is being modified.
```
IEnumerable<T> dependencies = TypesToImport.SelectMany(t => Data.GetDependencies(t)).Select(t => Data.TranslateAlias(t));
TypesToRemove = cleanedEntities.Where(e => !dependencies.Contains(e)).ToList();
```

Alternatively could the check be added above (another `entityClean`) step - then it just wouldn't be added in the first place.

Reply:
Thanks, I've gone with the approach suggested. I think it's safest to do this at the end so that the full list is iterated first in case there are two types with the same dependency, like `MaskedGoon2` and `MaskedGoon3`, who both depend on `MaskedGoon1`. Then we're not doing the same dependency check twice.